rust/worker/src/server.rs

```diff
@@ -243,19 +244,33 @@ impl WorkerServer {
         if request.limit.is_some() || request.offset.is_some() {
             return Err(Status::unimplemented("Limit and offset not supported"));
         }
-        if request.where_document.is_some() {
-            return Err(Status::unimplemented("Where document not supported"));
-        }
-        if request.r#where.is_some() {
-            return Err(Status::unimplemented("Where not supported"));
-        }
 
         // If no ids are provided, pass None to the orchestrator
         let query_ids = match request.ids.len() {
             0 => None,
             _ => Some(request.ids),
         };
 
+        let where_clause = match request.r#where {
+            Some(where_clause) => match where_clause.try_into() {
+                Ok(where_clause) => Some(where_clause),
+                Err(_) => return Err(Status::internal(format!("Error converting where clause",))),
+            },
+            None => None,
+        };
+
+        let where_document_clause = match request.where_document {
+            Some(where_document_clause) => match where_document_clause.try_into() {
+                Ok(where_document_clause) => Some(where_document_clause),
+                Err(_) => {
+                    return Err(Status::internal(format!(
```

Comment:
Same msg as above wrt tracing here as well as in other error places

Reply:
Done